packages/aztec.js/src/proof/joinSplit/index.js

```diff
@@ -219,33 +225,29 @@ joinSplit.encodeJoinSplitTransaction = ({
         publicOwner,
     );
 
-    const inputSignatures = inputNotes.map((inputNote, index) => {
-        const domain = signer.generateAZTECDomainParams(validatorAddress, constants.eip712.ACE_DOMAIN_PARAMS);
-        const schema = constants.eip712.JOIN_SPLIT_SIGNATURE;
-        const message = {
-            proof: proofs.JOIN_SPLIT_PROOF,
-            noteHash: inputNote.noteHash,
-            challenge,
-            sender: senderAddress,
-        };
-        const { privateKey } = inputNoteOwners[index];
-        const { signature } = signer.signTypedData(domain, schema, message, privateKey);
-        return signature;
-    });
+    if (inputNoteOwners.length !== 0 && validatorAddress.length !== 0) {
+        const signaturesArray = inputNotes.map((inputNote, index) => {
+            const domain = signer.generateZKAssetDomainParams(validatorAddress);
+            const schema = constants.eip712.JOIN_SPLIT_SIGNATURE;
+            const message = {
+                proof: proofs.JOIN_SPLIT_PROOF,
+                noteHash: inputNote.noteHash,
+                challenge,
+                sender: senderAddress,
+            };
+            const { privateKey } = inputNoteOwners[index];
+            const { signature } = signer.signTypedData(domain, schema, message, privateKey);
+            const concatenatedSignature = signature[0].slice(2) + signature[1].slice(2) + signature[2].slice(2);
+            return concatenatedSignature;
+        });
+        signatures = `0x${signaturesArray.join('')}`;
+    } else {
+        signatures = [];
```

Comment:
As discussed. 

Implementing a check for if `inputNoteOwners.length > 0 && validatorAddress.length === 0`, then throw. 

Otherwise, proceed. Replaced `inputNotes.map()` with `inputNoteOwners.map()`